internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -467,10 +466,17 @@ func (r *ClusterDeploymentsReconciler) createNewCluster(
 		return ctrl.Result{}, nil
 	}
 
+	openshiftVersion, err := r.addOpenshiftVersion(ctx, spec, pullSecret)
+	if err != nil {
+		r.Log.WithError(err).Error("failed to add OCP version")
```

Comment:
Why could this function fail? 
if you return `ctrl.Result{}, nil` then the resource will not be reconciled and the user will not know if something is wrong. 
Need to check the error type and react, update conditions and  if 4xx not requeue, 5xx requeue

Reply:
sure, used 'r.updateStatus' instead as done for other error handling